File: Backend/WebMedicalApplication/BackendMedicalApplication/DTo/UserDto.cs
namespace BackendMedicalApplication.DTo
{
    public class UserDto
    {
        public int UserId { get; set; }
        public string Username { get; set; }
        public string EmailAddress { get; set; }
        public int RoleId { get; set; }
        public string Name { get; set; }
        public string Surname { get; set; }
        public string CNP { get; set; }
        public int Age { get; set; }
        public string Password { get; set; }
        public string PhoneNumber { get; set; }
        public string Address { get; set; }
        public string? ResetPasswordCode { get; set; }
        public DateTime? ResetPasswordCodeExpires { get; set; }
    }
}  

File: Frontend/FrontMedicalApplication/src/app/core/dtos/user.dto.ts
import { ReviewDto } from './review.dto';

export class UserDto {
  userId?: number;
  username: string;
  password?: string;
  emailAddress: string;
  roleId: number;
  name: string;
  surname: string;
  cnp?: string;
  age?: number;
  address?: string;
  phoneNumber?: string;
  token?: string;
  description?: string;
  averageRating?: number = 0;
  reviewCount?: number = 0;
  reviews?: ReviewDto[] = [];

  constructor(
    username: string = '',
    emailAddress: string = '',
    roleId: number,
    name: string = '',
    surname: string = '',
    cnp?: string,
    age?: number,
    userId?: number,
    password?: string,
    address?: string,
    phoneNumber?: string,
    description?: string,
    token?: string
  ) {
    this.username = username;
    this.emailAddress = emailAddress;
    this.roleId = roleId;
    this.name = name;
    this.surname = surname;
    this.cnp = cnp;
    this.age = age;
    if (userId) {
      this.userId = userId;
    }
    this.password = password;
    this.address = address;
    this.phoneNumber = phoneNumber;
    this.description = description;
    this.token = token;
  }
}
